sysl2/sysl/cmd_runner.go

```diff
@@ -63,8 +63,35 @@ func (r *Runner) Configure(app *kingpin.Application) error {
 }
 
 // Temp until all the commands move in here
-func (r *Runner) HasCommand(which string) bool {
+func (r *cmdRunner) HasCommand(which string) bool {
 
 	_, ok := r.commands[which]
 	return ok
 }
+
+// Helper function to validate that a set of command flags are not empty values
+func EnsureFlagsNonEmpty(cmd *kingpin.CmdClause, _ ...string) {
+	fn := func(c *kingpin.ParseContext) error {
+
+		var errorMsg strings.Builder
+		for _, f := range cmd.Model().Flags {
+			val := f.Value.String()
+
+			if val != "" {
+				val = strings.Trim(val, " ")
+				if val == "" {
+					errorMsg.WriteString("'" + f.Name + "'" + " value passed is empty\n")
+				}
+			} else if len(f.Default) > 0 {
+				errorMsg.WriteString("'" + f.Name + "'" + " value passed is empty\n")
+			}
+		}
+		if errorMsg.Len() > 0 {
+			return errors.New(errorMsg.String())
+		}
+		return nil
+
+	}
+
+	cmd.PreAction(fn)
+}
```

Comment:
Kingpin provides OOB validation hooks at the application and the comand level. These hooks are automaticalIy invoked when the command line is parsed. I suggest using them instead.  
type CmdClauseValidator func(*CmdClause) error

Reply:
I tried that, cant remember why it didnt work though, will try it again.

Reply:
@springwiz yeah, just checked again, validate appaears to happen at the wrong point in the parsing, PreAction works better.